hubspot_sync/tasks.py

```diff
@@ -134,15 +136,15 @@ def handle_failed_batch_chunk(chunk: list[int], hubspot_type: str) -> list[int]:
 @single_task(10, key=task_obj_lock)
 def sync_contact_with_hubspot(user_id: int) -> str:
     """
-    Sync a user with a hubspot contact
+    Sync a User with a hubspot contact
 
     Args:
-        user_id(int): The User id
+        user_id(int): The User ID.
 
     Returns:
-        str: The hubspot id for the contact
+        bool: True if the HubSpot contact was successfully updated, otherwise False.
     """
-    return api.sync_contact_with_hubspot(user_id).id
+    return api.sync_contact_with_hubspot(User.objects.get(id=user_id))
```

Comment:
```python
return api.sync_contact_with_hubspot(User.objects.get(id=user_id)).id
```

Otherwise this exception is thrown by celery:
```
celery_1    | [2023-07-11 16:32:09,309: ERROR/ForkPoolWorker-8] Task hubspot_sync.tasks.sync_contact_with_hubspot[441305cd-039c-4bfb-afd7-bec68899f2af] raised unexpected: EncodeError(TypeError('Object of type SimplePublicObject is not JSON serializable'))
celery_1    | Traceback (most recent call last):
celery_1    |   File "/usr/local/lib/python3.9/site-packages/kombu/serialization.py", line 39, in _reraise_errors
celery_1    |     yield
celery_1    |   File "/usr/local/lib/python3.9/site-packages/kombu/serialization.py", line 210, in dumps
celery_1    |     payload = encoder(data)
celery_1    |   File "/usr/local/lib/python3.9/site-packages/kombu/utils/json.py", line 68, in dumps
celery_1    |     return _dumps(s, cls=cls or _default_encoder,
celery_1    |   File "/usr/local/lib/python3.9/json/__init__.py", line 234, in dumps
celery_1    |     return cls(
celery_1    |   File "/usr/local/lib/python3.9/json/encoder.py", line 199, in encode
celery_1    |     chunks = self.iterencode(o, _one_shot=True)
celery_1    |   File "/usr/local/lib/python3.9/json/encoder.py", line 257, in iterencode
celery_1    |     return _iterencode(o, 0)
celery_1    |   File "/usr/local/lib/python3.9/site-packages/kombu/utils/json.py", line 58, in default
celery_1    |     return super().default(o)
celery_1    |   File "/usr/local/lib/python3.9/json/encoder.py", line 179, in default
celery_1    |     raise TypeError(f'Object of type {o.__class__.__name__} '
celery_1    | TypeError: Object of type SimplePublicObject is not JSON serializable
celery_1    |
celery_1    | During handling of the above exception, another exception occurred:
celery_1    |
celery_1    | Traceback (most recent call last):
celery_1    |   File "/usr/local/lib/python3.9/site-packages/celery/app/trace.py", line 518, in trace_task
celery_1    |     task.backend.mark_as_done(
celery_1    |   File "/usr/local/lib/python3.9/site-packages/celery/backends/base.py", line 162, in mark_as_done
celery_1    |     self.store_result(task_id, result, state, request=request)
celery_1    |   File "/usr/local/lib/python3.9/site-packages/celery/backends/base.py", line 528, in store_result
celery_1    |     self._store_result(task_id, result, state, traceback,
celery_1    |   File "/usr/local/lib/python3.9/site-packages/celery/backends/base.py", line 962, in _store_result
celery_1    |     self._set_with_state(self.get_key_for_task(task_id), self.encode(meta), state)
celery_1    |   File "/usr/local/lib/python3.9/site-packages/celery/backends/base.py", line 418, in encode
celery_1    |     _, _, payload = self._encode(data)
celery_1    |   File "/usr/local/lib/python3.9/site-packages/celery/backends/base.py", line 422, in _encode
celery_1    |     return dumps(data, serializer=self.serializer)
celery_1    |   File "/usr/local/lib/python3.9/site-packages/kombu/serialization.py", line 210, in dumps
celery_1    |     payload = encoder(data)
celery_1    |   File "/usr/local/lib/python3.9/contextlib.py", line 135, in __exit__
celery_1    |     self.gen.throw(type, value, traceback)
celery_1    |   File "/usr/local/lib/python3.9/site-packages/kombu/serialization.py", line 43, in _reraise_errors
celery_1    |     reraise(wrapper, wrapper(exc), sys.exc_info()[2])
celery_1    |   File "/usr/local/lib/python3.9/site-packages/kombu/exceptions.py", line 21, in reraise
celery_1    |     raise value.with_traceback(tb)
celery_1    |   File "/usr/local/lib/python3.9/site-packages/kombu/serialization.py", line 39, in _reraise_errors
celery_1    |     yield
celery_1    |   File "/usr/local/lib/python3.9/site-packages/kombu/serialization.py", line 210, in dumps
celery_1    |     payload = encoder(data)
celery_1    |   File "/usr/local/lib/python3.9/site-packages/kombu/utils/json.py", line 68, in dumps
celery_1    |     return _dumps(s, cls=cls or _default_encoder,
celery_1    |   File "/usr/local/lib/python3.9/json/__init__.py", line 234, in dumps
celery_1    |     return cls(
celery_1    |   File "/usr/local/lib/python3.9/json/encoder.py", line 199, in encode
celery_1    |     chunks = self.iterencode(o, _one_shot=True)
celery_1    |   File "/usr/local/lib/python3.9/json/encoder.py", line 257, in iterencode
celery_1    |     return _iterencode(o, 0)
celery_1    |   File "/usr/local/lib/python3.9/site-packages/kombu/utils/json.py", line 58, in default
celery_1    |     return super().default(o)
celery_1    |   File "/usr/local/lib/python3.9/json/encoder.py", line 179, in default
celery_1    |     raise TypeError(f'Object of type {o.__class__.__name__} '
celery_1    | kombu.exceptions.EncodeError: Object of type SimplePublicObject is not JSON serializable
```